src/core/public/http/types.ts

```diff
@@ -217,30 +217,56 @@ export interface HttpFetchOptions extends HttpRequestInit {
   headers?: HttpHeadersInit;
 
   /**
-   * When `true` the return type of {@link HttpHandler} will be an {@link IHttpResponse} with detailed request and
+   * Whether or not the request should include the "system API" header to differentiate an end user request from Kibana
+   * internal request.
+   * Can be read on the server-side using KibanaRequest#isSystemApi. Defaults to `false`.
+   */
+  asSystemApi?: boolean;
+
+  /**
+   * When `true` the return type of {@link HttpHandler} will be an {@link HttpResponse} with detailed request and
    * response information. When `false`, the return type will just be the parsed response body. Defaults to `false`.
    */
   asResponse?: boolean;
 }
 
+/**
+ * Similar to {@link HttpFetchOptions} but with the URL path included.
+ * @public
+ */
+export interface HttpFetchOptionsWithPath extends HttpFetchOptions {
+  /*
+   * The path on the Kibana server to send the request to. Should not include the basePath.
+   */
+  path: string;
+}
+
 /**
  * A function for making an HTTP requests to Kibana's backend. See {@link HttpFetchOptions} for options and
- * {@link IHttpResponse} for the response.
+ * {@link HttpResponse} for the response.
  *
  * @param path the path on the Kibana server to send the request to. Should not include the basePath.
  * @param options {@link HttpFetchOptions}
- * @returns a Promise that resolves to a {@link IHttpResponse}
+ * @returns a Promise that resolves to a {@link HttpResponse}
  * @public
  */
 export interface HttpHandler {
+  /** @deprecated */
   <TResponseBody = any>(path: string, options: HttpFetchOptions & { asResponse: true }): Promise<
-    IHttpResponse<TResponseBody>
+    HttpResponse<TResponseBody>
+  >;
+  <TResponseBody = any>(options: HttpFetchOptionsWithPath & { asResponse: true }): Promise<
+    HttpResponse<TResponseBody>
   >;
+  /** @deprecated */
   <TResponseBody = any>(path: string, options?: HttpFetchOptions): Promise<TResponseBody>;
+  <TResponseBody = any>(options: HttpFetchOptionsWithPath): Promise<TResponseBody>;
 }
 
 /** @public */
-export interface IHttpResponse<TResponseBody = any> {
+export interface HttpResponse<TResponseBody = any> {
```

Comment:
__question:__ out of curiosity, we don't follow `I{interface_name}` name convention anymore?

Reply:
We use the `I-` prefix convention to distinguish classes from interfaces. https://github.com/elastic/kibana/pull/52397/files#diff-80ee07cd5df667d1b21f1f90416adf2dR32-R33
We are going to abandon this convention with TS v3.8 release. TS should introduce support for private fields.

Reply:
I see, thanks. Okay, so if we have similar issues in plugins it'd better to use `PublicMethodOf` it seems (clearer intention and easier to migrate when we switch to TS 3.8).